File: apps/sdkwork-knowledgebase-h5/src/bootstrap/runtime.ts
import { resolveKnowledgebaseH5Environment } from "./environment";
import { registerHostAdapters } from "./hostAdapters";
import { createKnowledgebaseH5IamRuntime } from "./iamRuntime";
import { bootstrapSdkClients } from "./sdkClients";

/**
 * Application bootstrap.
 *
 * Assembles providers, routes, and shell wiring only. Business screens,
 * services, and state live in `packages/`.
 */
export function bootstrap() {
  const environment = resolveKnowledgebaseH5Environment();
  const iamRuntime = createKnowledgebaseH5IamRuntime();
  const sdkClients = bootstrapSdkClients();
  const host = registerHostAdapters();

  return { environment, host, iamRuntime, sdkClients };
}

File: apps/sdkwork-knowledgebase-flutter-mobile/lib/bootstrap/runtime.dart
import 'host_adapters.dart';
import 'iam_runtime.dart';
import 'routes.dart';
import 'sdk_clients.dart';

/// Root composition result.
///
/// The root assembles environment resolution, SDK clients, IAM runtime, host
/// adapters, and the route registry only. It owns no product business logic.
class KnowledgebaseMobileRuntime {
  const KnowledgebaseMobileRuntime({required this.sdkClients});

  final SdkClients sdkClients;
}

Future<KnowledgebaseMobileRuntime> bootstrap() async {
  createIamRuntime();
  registerHostAdapters();
  final sdkClients = createSdkClients();
  createRoutes();
  return KnowledgebaseMobileRuntime(sdkClients: sdkClients);
}
